opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -108,3 +111,73 @@ def _create_immutable_attributes(
     attributes: types.Attributes,
 ) -> types.Attributes:
     return MappingProxyType(attributes.copy() if attributes else {})
+
+
+_DEFAULT_LIMIT = 128
+
+
+class BoundedDict(MutableMapping):
+    """An ordered dict with a fixed max capacity.
+
+    Oldest elements are dropped when the dict is full and a new element is
+    added.
+    """
+
+    def __init__(
+        self,
+        maxlen: Optional[int] = _DEFAULT_LIMIT,
+        attributes: types.Attributes = None,
+        immutable: bool = True,
+    ):
+        if maxlen is not None:
+            if not isinstance(maxlen, int):
+                raise ValueError
+            if maxlen < 0:
+                raise ValueError
```

Comment:
A proper error message would be preferred.

Reply:
Added `maxlen must be valid int greater or equal to 0` message to value error.